src/detectors/LOWQ2/LOWQ2.cc

```diff
@@ -30,6 +34,63 @@ extern "C" {
          app
     ));
 
+    // This should really be done before digitization as summing hits in the same cell couldn't evet be mixed between layers. At the moment just prep for clustering.
+    std::vector<int> moduleIDs{1,2};
+    std::vector<int> layerIDs {0,1,2,3};
+    std::vector<std::vector<int>> segmentIDs{};
+    std::vector<std::string> segmentDiv;
+
+    for(int mod_id : moduleIDs){
+      for(int lay_id : layerIDs){
+        segmentIDs.push_back({mod_id,lay_id});
+        segmentDiv.push_back(fmt::format("TaggerTrackerM{}L{}RawHits",mod_id,lay_id));
+      }
+    }
+
+    app->Add(new JOmniFactoryGeneratorT<SplitGeometry_factory<edm4eic::RawTrackerHit>>(
+         "TaggerTrackerSplitHits",
+         {"TaggerTrackerRawHits"},
+         segmentDiv,
+         {
+           .divisions = segmentIDs,
+           .readout   = "TaggerTrackerHits",
+           .division  = {"module","layer"},
+         },
+         app
+      )
+    );
+
+    std::vector<std::string> layerClusters;
+
+    // Clustering of hits in each layer
+    for(int mod_id : moduleIDs){
+      for(int lay_id : layerIDs){
```

Comment:
How expensive is splitting into the graph nodes? I feel this ideally would be split prior to digitization and remain split until tracks are made to avoid lots of unnecessary combinatoric comparisons. Although for some reason I'm apposed to splitting it up in the simulation readout.

I was considering dropping this clustering step in the analysis for the time being anyway. A geant4 hit is only associated to a single segmentation cell even if the track passes through several and there is no charge sharing implemented at the moment. As far as I'm aware this is also the only place using the ROOT RVec which is very convenient but might not be in line with the philosophy.

Reply:
Alternatively to splitting the computational graph, the input and output collection list could both be variadic and looping over the collections within the algorithm. Would that be a neater solution?

Reply:
I don't know how bad are the combinatorics in this case. If this is done purely for optimization, this might be premature. Ideally, we go for the MVP first and add complexity as needed.

Regarding not needing clustering, let me know if we should skip this and review the linear fitting PR instead..

Reply:
Might as well get it in if it's almost ready. It could provide a nice example to actually see the effects of implementing the charge sharing. I've swapped to looping over the collections inside the algorithm, although the other approach should still work too.

I realised I was just merging the clusters again at the bottom which was meant to originally test that the merger worked I've now removed that.